### PR TITLE
feat(run): Include the workspace root .bin in env path

### PR DESCRIPTION
**Pull request by rally25rs**
**Summary**

Partial resolution for #4543.

Previously, when running `yarn run` the env PATH would be set to look in node_modules/.bin, however, in workspaces the root workspace .bin path was not being included.

This PR adds the workspace root
node_modules/.bin path after the individual package's path.

This is generally needed because #4730 ensures bin links in a workspace will be at the workspace root. With this PR, you can now `yarn run` commands in an individual package again.

**Test plan**

Manually tested by adding a script that runs `echo $PATH`

### CODE REVIEW

**Comment by BYK** [comment]
This is awesome! Let's add one more test, merge from master so we get the latest CI upgrades and I'm game!

**Comment by rally25rs** [comment]
@BYK finally got around to fixing this test. All ✅ now 👍

File: __tests__/commands/run.js [hunk 1]
```diff
@@ -133,3 +144,20 @@ test('adds quotes if args have spaces and quotes', (): Promise<void> => {
...
   });
 });
+
+test('adds workspace root node_modules/.bin to path when in a workspace', (): Promise<void> => {
+  return runRunInWorkspacePackage('packages/pkg1', ['env'], {}, 'workspace', (config, reporter): ?Promise<void> => {
+    const logEntry = reporter.getBuffer().find(entry => entry.type === 'log');
+    const logData = logEntry ? logEntry.data.toString() : '{}';
+    const parsed = logEntry ? JSON.parse(logData) : {};
+    let envPaths = [];
+    if (parsed.PATH) {
```
Comment (BYK) [inline __tests__/commands/run.js]: Please use [`path.delimiter`](https://nodejs.org/api/path.html#path_path_delimiter) instead. I'll go ahead and fix this myself and then merge tho :)

File: __tests__/commands/run.js [hunk 2]
```diff
@@ -133,3 +144,20 @@ test('adds quotes if args have spaces and quotes', (): Promise<void> => {
...
+      envPaths = parsed.PATH.split(':');
+    } else if (parsed.Path) {
+      envPaths = parsed.Path.split(';');
+    }
+
+    expect(envPaths).toContain(path.join(config.cwd, 'node_modules/.bin'));
```
Comment (BYK) [inline __tests__/commands/run.js]: I expected this to fail on Windows since the path separator should be `\` there. Weird.

Comment (rally25rs) [reply __tests__/commands/run.js]: IIRC `path.join()` converts it auto-magically.

File: __tests__/commands/run.js (hunk 1 above)
Comment (rally25rs) [reply __tests__/commands/run.js]: Oh cool, I didn't even know that was a thing. Thanks!

File: __tests__/commands/run.js (hunk 2 above)
Comment (rally25rs) [reply __tests__/commands/run.js]: although... why didn't I do `path.join(config.cwd, 'node_modules', '.bin')`? 😕 ...I'm an idiot 😆

Comment (BYK) [reply __tests__/commands/run.js]: Wow, nice.

**Comment by iansu** [comment]
Any idea when this is going to be released?

**Comment by jonaskello** [comment]
I cannot get this to work. Is it included in 1.3.2?

**Comment by simonbuchan** [comment]
I think there was some talk about putting the release tags a merged PR is included by, like they are in commits, but for now you can check in the footer of the commit message on the page for the merge commit, which is linked in the merge message above.

**Pull request merged**